tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -2103,39 +2103,32 @@ Status FragmentMetadata::load_tile_var_sizes(
 /*        PRIVATE METHODS         */
 /* ****************************** */
 
-Status FragmentMetadata::get_footer_size(
-    uint32_t version, uint64_t* size) const {
-  if (version < 3) {
-    *size = footer_size_v3_v4();
-  } else if (version < 4) {
-    *size = footer_size_v5_v6();
-  } else if (version < 11) {
-    *size = footer_size_v7_v10();
-  } else {
-    *size = footer_size_v11_or_higher();
-  }
-
-  return Status::Ok();
-}
-
 uint64_t FragmentMetadata::footer_size() const {
   return footer_size_;
 }
 
 Status FragmentMetadata::get_footer_offset_and_size(
     uint64_t* offset, uint64_t* size) const {
+  uint32_t old_f_version;
   uint32_t f_version;
   auto name = fragment_uri_.remove_trailing_slash().last_path_part();
-  RETURN_NOT_OK(utils::parse::get_fragment_name_version(name, &f_version));
-  if (array_schema_->domain().all_dims_fixed() && f_version < 5) {
-    RETURN_NOT_OK(get_footer_size(f_version, size));
+  RETURN_NOT_OK(utils::parse::get_fragment_name_version(name, &old_f_version));
+  RETURN_NOT_OK(utils::parse::get_fragment_version(name, &f_version));
+  std::cerr << "Old f_version: " << old_f_version << " New: " << f_version << std::endl;
+  if (f_version < 3) {
```

Comment:
This seems like a change in behavior... We don't have an extensive set of very old arrays to test so it's something we should be careful about. I'm not sure the fragment version will come in correctly for old arrays so I'd rather keep the behavior where the fragment name version is used for versions smaller than 5.

Reply:
Yeah, my initial bug fix was quite broken. The mismatch in versions is a bit weirder than I initially thought. I'm currently abusing CI to try and debug what the original behavior was so that I can specifically replicate it with a complete writeup so that we're sure its backwards compatible.